### PR TITLE
Web Development Related Books URLs

### PR DESCRIPTION
**Pull request by haziq-bangash**
## What does this PR do?
Add resource(s)

### Description
Added Books that have important notes on Node js, React js and Typescript

### Why is this valuable (or not)?
If someone is going for MERN stack Development, they might find these useful.

### How do we know it's really free?
It is a website that have free books resources regarding programming languages. [Link](https://books.goalkicker.com/) to the website.

### For book lists, is it a book? For course lists, is it a course? etc.
It is a Book

## Checklist:
- [x]  Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).

- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

File: books/free-programming-books-subjects.md [hunk 1]
```diff
@@ -875,7 +875,10 @@ Kerridge (PDF) (email address *requested*, not required)
 
 ### Web Services
 
+* [Node.js Notes for Professionals](https://books.goalkicker.com/NodeJSBook/)
+* [React JS Notes for Professionals](https://books.goalkicker.com/ReactJSBook/)
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:879]: Already in books

Comment (davorpa) [inline books/free-programming-books-subjects.md:879]: Remove duplicated resources. Already exists in `books-langs.md` file

```suggestion
```

File: books/free-programming-books-subjects.md [hunk 2]
```diff
@@ -876,6 +876,7 @@ Kerridge (PDF) (email address *requested*, not required)
 ### Web Services
 
 * [RESTful Web Services](http://restfulwebapis.org/RESTful_Web_Services.pdf) (PDF)
+* [TypeScript Notes for Professionals](https://books.goalkicker.com/TypeScriptBook2/)
```
Comment (davorpa) [inline books/free-programming-books-subjects.md:879]: Could yo move this to `books-langs`? You select a wrong category

**Review by davorpa — CHANGES_REQUESTED** [review]

File: books/free-programming-books-subjects.md (hunk 2 above)
Comment (haziq-bangash) [reply books/free-programming-books-subjects.md:879]: yes

File: books/free-programming-books-langs.md [hunk 3]
```diff
@@ -1109,6 +1109,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [The JavaScript Way](https://github.com/bpesquet/thejsway) - Baptiste Pesquet
 * [The Problem with Native JavaScript APIs](https://www.oreilly.com/programming/free/native-javascript-apis.csp) - Nicholas C. Zakas (PDF, email address *requested*)
 * [Thinking in JavaScript](https://www.amazon.com/Thinking-JavaScript-Aravind-Shenoy-ebook/dp/B00JUI6LUQ) - Aravind Shenoy (Kindle)
+* [TypeScript Notes for Professionals](https://books.goalkicker.com/TypeScriptBook2/)
```
Comment (davorpa) [inline books/free-programming-books-langs.md:1112]: Adding creators and format notes

```suggestion
* [TypeScript Notes for Professionals](https://books.goalkicker.com/TypeScriptBook2/) - Compiled from StackOverflow Documentation (PDF)
```

**Review by davorpa — APPROVED** [review]

File: books/free-programming-books-langs.md [hunk 4]
```diff
@@ -1109,6 +1109,7 @@ That section got so big, we decided to split it into its own file, the [BY SUBJE
 * [The JavaScript Way](https://github.com/bpesquet/thejsway) - Baptiste Pesquet
 * [The Problem with Native JavaScript APIs](https://www.oreilly.com/programming/free/native-javascript-apis.csp) - Nicholas C. Zakas (PDF, email address *requested*)
 * [Thinking in JavaScript](https://www.amazon.com/Thinking-JavaScript-Aravind-Shenoy-ebook/dp/B00JUI6LUQ) - Aravind Shenoy (Kindle)
+* [TypeScript Notes for Professionals](https://books.goalkicker.com/TypeScriptBook2/) - Compiled from StackOverflow Documentation (PDF)
```
Comment (eshellman) [inline books/free-programming-books-langs.md:1112]: Why is this not in the typescript section?

Comment (davorpa) [reply books/free-programming-books-langs.md:1112]: Was my mistake :worried: don't check the section after movement. Solving...

**Review by davorpa — APPROVED** [review]

**Comment by eshellman** [comment]
added elsewhere

**Pull request closed without merging**